code/clue.js

```diff
@@ -1,59 +1,326 @@
+  age: 60,
+  image: "assets/mustard.png",
+  occupation: "Colonel",
+  favouriteWeapon: "poison"
+};
+
+const mrsWhite = {
+  firstName: "Mrs",
+  lastName: "White",
+  color: "white",
+  description: "Widow of a nuclear physicist",
+  age: 62,
+  image: "assets/white.png",
+  occupation: "Servant",
+  favouriteWeapon: "axe"
+};
 
-// CREATE OBJECTS FOR ALL THE WEAPONS, ADD MORE CHARACTERISTICS TO THE WEAPONS IF YOU LIKE.
+// OBJECTS FOR ALL THE WEAPONS 
 
 const rope = {
-  name: 'rope',
-  weight: 10
-}
+  name: "rope",
+  image: "assets/rope.png",
+  weight: 10,
+  id : "rope"
+};
+
+const knife = {
+  name: "knife",
+  image: "assets/knife.png",
+  weight: 1,
+  id: "knife"
+};
+
+const candlestick = {
+  name: "candlestick",
+  image: "assets/candlestick.png",
+  weight: 1,
+  id: "candlestick"
+};
+
+const bomb = {
+  name: "bomb",
+  image: "assets/bomb.png",
+  weight: 5,
+  id: "bomb"
+};
+
+const poison = {
+  name: "poison",
+  image: "assets/poison.png",
+  weight: 1,
+  id: "poison"
+};
+
+const axe = {
+  name: "axe",
+  image: "assets/axe.png",
+  weight: 7,
+  id: "axe"
+};
 
-// THE ROOMS ONLY HAS A NAME SO NO NEED FOR OBJECTS THERE.
+const bat = {
+  name: "bat",
+  image: "assets/bat.png",
+  weight: 4,
+  id: "bat"
+};
 
-// NOW GROUP ALL SUSPECTS, WEAPONS AND ROOMS IN ARRAYS LIKE THIS:
+const wench = {
+  name: "wench",
+  image: "assets/wench.png",
+  weight: 8,
+  id: "wench"
+};
+
+const pistol = {
+  name: "pistol",
+  image: "assets/pistol.png",
+  weight: 3,
+  id: "pistol"
+};
+
+// Array for suspects, weapons, rooms (rooms is not object)
 
 const suspects = [
   mrGreen,
-  mrsWhite
-  // ...  and the rest
-]
+  prPlum,
+  missScarlett,
+  mrsPeacock,
+  colMustard,
+  mrsWhite,
+];
+
+const weapons = [
+  rope,
+  knife,
+  candlestick,
+  bomb,
+  poison,
+  axe,
+  bat,
+  wench,
+  pistol,
+];
+
+const rooms = [
+  "Dining Room",
+  "Conservatory",
+  "Kitchen",
+  "Study",
+  "Library",
+  "Billiard Room",
+  "Lounge",
+  "Ballroom",
+  "Hall",
+  "Spa",
+  "Living Room",
+  "Observatory",
+  "Theater",
+  "Guest House",
+  "Patio"
+];
+
+// Random selector, takes an array and returns one random item in the array
+
+const randomSelector = (array) => {
+  return array[Math.floor(Math.random() * array.length)];
+};
 
-const weapons = []
+// Mystery object 
 
-const rooms = []
+const mystery = {
+  killer : null, 
+  weapon : null,  
+  room: null
+};
 
-// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
-const randomSelector = array => {
-  return array[Math.floor(Math.random() * array.length)]
+// Progress bar for loading cards. Bar will oly be shown once clicked. After animation completes, callback function is called for picking
+// killer.
+
+let i = 0;
+const move = (callback) => {
+  document.getElementById('myProgress').style.display = "block";
+  if (i == 0) {
```

Comment:
in 99% of cases you want to use `===` instead of `==` - the second one is checking only value, first one is checking both value and type, so :
`12 == "12"` is `true`
`12 === "12"` is `false`

Reply:
Needs to be solved

Reply:
fixed